### PR TITLE
openssl: enable z/OS support

### PR DESCRIPTION
**Pull request by IgorTodorovskiIBM**
Enable building of openssl and openssl-cl for z/OS:
* Add no_asm configurations for z/OS (z/OS does not have asm at this point - will change once we get GSKit upstreamed/building)
* Add gyp configuration cflags and defines for z/OS
* Add getentropy guard and function call for z/OS
* Define NO_SYS_PARAM_H for z/OS in sockets.h

### CODE REVIEW

**Review by zsw007 — APPROVED** [review]
LGTM

**Comment by alexcfyung** [comment]
some of the checks in this and previous pull request always seems to fail, do we need to care about those?

**Comment by gabylb** [comment]
For lint failure on my PR it was valid (complaining about lines with only blank spaces).
The above still shows one case, though it's likely an older copy of configure.py.
Igor can you re-run the job for lint (apparently it doesn't allow me to).

File: deps/openssl/config/archs/zos-s390x/no-asm/crypto/include/internal/bn_conf.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Copyright 2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef HEADER_BN_CONF_H
+# define HEADER_BN_CONF_H
```
Comment (gabylb) [inline deps/openssl/config/archs/zos-s390x/no-asm/crypto/include/internal/bn_conf.h:13]: Suggest to remove the blank space in '# define' as well as in the other lines below.
https://developer.gnome.org/programming-guidelines/stable/c-coding-style.html.en

Comment (gabylb) [reply deps/openssl/config/archs/zos-s390x/no-asm/crypto/include/internal/bn_conf.h:13]: If the #if section is more than a few lines, the `#endif` is usually followed by something like `  // HEADER_BN_CONF_H`

Comment (IgorTodorovskiIBM) [reply deps/openssl/config/archs/zos-s390x/no-asm/crypto/include/internal/bn_conf.h:13]: Unfortunately, this is a file generated from `deps/openssl/openssl/include/crypto/bn_conf.h.in.`  The other platforms have this as well, for example: `source/node/deps/openssl/config/archs/linux64-s390x/no-asm/crypto/include/internal/bn_conf.h`

**Comment by IgorTodorovskiIBM** [comment]
Not sure why the checks are failing, they aren't related to this commit and Gaby's PR fixes the configure.py so this is likely using the older configure.py

**Review by gabylb — APPROVED** [review]
LGTM

**Review by alexcfyung — APPROVED** [review]
LGTM

**Pull request merged**